### PR TITLE
Re-enable assertion on operator role mappings

### PR DESCRIPTION
**Pull request by thbkrkr**
This re-enables the assertion on operator-level role mappings in e2e test TestStackConfigPolicy.

### CODE REVIEW

**Comment by thbkrkr** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy -m s=8.10.4,s=8.11.4,s=8.12.2,s=8.13.4,s=8.14.3,s=8.15.1,s=8.16.0-SNAPSHOT

**Comment by thbkrkr** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy -m s=8.14.3,s=8.15.1,s=8.16.0-SNAPSHOT

**Comment by thbkrkr** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy -m s=8.14.3,s=8.15.1,s=8.16.0-SNAPSHOT

File: test/e2e/es/stackconfigpolicy_test.go [hunk 1]
```diff
@@ -67,15 +67,30 @@ func TestStackConfigPolicy(t *testing.T) {
...
-		"/_index_template/template_test",
-		"/_component_template/runtime_component_template_test",
-		"/_component_template/component_template_test",
+	configuredObjectsEndpoints := map[string]statusCode{
+		"/_snapshot/repo_test":  {ok, notFound},
+		"/_slm/policy/slm_test": {ok, notFound},
+		"/_security/role_mapping/role_test": {
+			onCreation: func() int {
+				// due to a bug in 8.15.x, role mappings are not found via the API
+				if stackVersion.GTE(version.MinFor(8, 15, 0)) && stackVersion.LT(version.MinFor(8, 16, 0)) {
```
Comment (thbkrkr) [inline test/e2e/es/stackconfigpolicy_test.go:76]:
```suggestion
				if stackVersion.GTE(version.MinFor(8, 15, 0)) && stackVersion.LT(version.MinFor(8, 15, 3)) {
```

**Comment by thbkrkr** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy -m s=8.14.3,s=8.15.1,s=8.15.2,s=8.15.3-SNAPSHOT,s=8.16.0-SNAPSHOT

File: test/e2e/es/stackconfigpolicy_test.go [hunk 2]
```diff
@@ -67,15 +67,30 @@ func TestStackConfigPolicy(t *testing.T) {
...
+				return ok()
+			},
+			onDeletion: func() int {
+				// prior to 8.15.x, role mappings are not deleted when scp is deleted
+				if stackVersion.LT(version.MinFor(8, 15, 0)) {
+					return ok()
+				}
+				return notFound()
+			},
+		},
```
Comment (thbkrkr) [inline test/e2e/es/stackconfigpolicy_test.go:88]: Thinking back, I think I should move the role mapping out of this map and do a specific validation for it.

Comment (pebrc) [reply test/e2e/es/stackconfigpolicy_test.go:88]: You could do a custom thing where we inspect cluster state instead. It is not considered a stable interface but might be still be OK to see if the role mappings are there in a test. As we know now that does not guarantee that they are also applied (at least no in the buggy versions)

**Comment by thbkrkr** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy -m s=8.14.3,s=8.15.1,s=8.15.2,s=8.15.3-SNAPSHOT,s=8.16.0-SNAPSHOT

File: test/e2e/es/stackconfigpolicy_test.go (hunk 2 above)
Comment (thbkrkr) [reply test/e2e/es/stackconfigpolicy_test.go]: I rewrote a new version in 3149c08 that uses `/_cluster/state/metadata?filter_path=metadata.role_mappings`.

**Comment by thbkrkr** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy -m s=8.14.3,s=8.15.1,s=8.15.2,s=8.15.3-SNAPSHOT,s=8.16.0-SNAPSHOT

**Comment by thbkrkr** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy -m s=8.14.3,s=8.15.1,s=8.15.2,s=8.15.3-SNAPSHOT,s=8.16.0-SNAPSHOT

**Comment by thbkrkr** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy -m s=8.14.3,s=8.15.1,s=8.15.2,s=8.15.3-SNAPSHOT,s=8.16.0-SNAPSHOT

**Comment by thbkrkr** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy -m s=8.14.3,s=8.15.1,s=8.15.2,s=8.15.3-SNAPSHOT,s=8.16.0-SNAPSHOT,s=8.17.0-SNAPSHOT

**Review by pebrc — APPROVED** [review]
LGTM

**Pull request merged**